tasmota/tasmota_xdrv_driver/xdrv_27_esp32_shutter.ino

```diff
@@ -528,6 +532,20 @@ uint16_t ShutterGetCycleTime(uint8_t i, uint8_t  max_runtime) {
   return cycle_time;
 }
 
+uint8_t ShutterGetFreeChannel() {
+  uint8_t nextFreeChannel = 0;
+  for (uint8_t i = 0; i < MAX_SHUTTERS_ESP32; i++) {
+    //SOC_LEDC_CHANNEL_NUM   
+    nextFreeChannel = tmax(nextFreeChannel, Shutter[i].ledc_channel);
+  }
+  if (nextFreeChannel == SOC_LEDC_CHANNEL_NUM) {
```

Comment:
I would do ``if (nextFreeChannel >= SOC_LEDC_CHANNEL_NUM) {`` as you never know where the user comes from.